### PR TITLE
Drop deprecated proc-macro-hack

### PR DESCRIPTION
**Pull request by VirxEC**
These changes do up the actual MSRV to at least Rust 1.45, but that was released 3 years ago on `July 16, 2020` so it shouldn't be an issue. I did a version bump anyway due to the MSRV bump and the updated 2021 edition is more constricting anyway.

From the `proc-macro-hack` crate:
> As of Rust 1.45 this crate is superseded by native support for #[proc_macro] in expression position. Only consider using this crate if you care about supporting compilers between 1.31 and 1.45.

Dropping this crate and using the std `#[proc_macro]` lets it build on stable Rust by default, without flags. The `stable` feature is still included due to the flags in the test modules, but has no actual impact on the code - I'm not sure what to do about it, but you _can_ remove the `stable` feature and everything works if you delete the requirement of the nightly only feature in the tests.

+ Dropped `proc-macro-hack`
+ Updated to the latest edition, `2021` (not required to compile on stable, could revert)
+ Updated `libflate` to 2.0 (not required to compile on stable, could revert)
+ Bumped version to `0.3.0` due to MSRV bump

### CODE REVIEW

File: tests/base64-raw.rs [hunk 1]
```diff
@@ -19,5 +19,5 @@ pub static DATA: &str = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/asset
 
 #[test]
 fn test() {
-    verify_str("base64.txt", &DATA);
+    verify_str("base64.txt", DATA);
```
Comment (SOF3) [inline tests/base64-raw.rs:22]: This change is unrelated to the proc macro hack stuff, just normal code cleanup right?

**Review by SOF3 — APPROVED** [review]

File: codegen-exports/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,2 @@
-#[cfg_attr(feature = "stable", proc_macro_hack::proc_macro_hack)]
 pub use include_flate_codegen::deflate_file;
-
-#[cfg_attr(feature = "stable", proc_macro_hack::proc_macro_hack)]
 pub use include_flate_codegen::deflate_utf8_file;
```
Comment (SOF3) [inline codegen-exports/src/lib.rs:2]: Could we just as well delete this reexport crate directly?

File: tests/base64-raw.rs (hunk 1 above)
Comment (VirxEC) [reply tests/base64-raw.rs:22]: yes, it was the only clippy warn I was getting so I just cleaned it up rq

File: codegen-exports/src/lib.rs (hunk 2 above)
Comment (VirxEC) [reply codegen-exports/src/lib.rs:2]: Honestly, I'm not sure but I would guess so.  Would just have to try it.

(The original comment was made too quickly, I didn't read it and thought this was about something else)

Comment (VirxEC) [reply codegen-exports/src/lib.rs:2]: #23
